nomad/job_endpoint_statuses.go

```diff
@@ -0,0 +1,303 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package nomad
+
+import (
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/armon/go-metrics"
+	"github.com/hashicorp/go-memdb"
+	"github.com/hashicorp/go-set/v2"
+	"github.com/hashicorp/nomad/acl"
+	"github.com/hashicorp/nomad/nomad/state"
+	"github.com/hashicorp/nomad/nomad/state/paginator"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+// Statuses looks up info about jobs, their allocs, and latest deployment.
+func (j *Job) Statuses(
+	args *structs.JobStatusesRequest,
+	reply *structs.JobStatusesResponse) error {
+
+	authErr := j.srv.Authenticate(j.ctx, args)
+	if done, err := j.srv.forward("Job.Statuses", args, args, reply); done {
+		return err
+	}
+	j.srv.MeasureRPCRate("jobs", structs.RateMetricList, args)
+	if authErr != nil {
+		return structs.ErrPermissionDenied
+	}
+	defer metrics.MeasureSince([]string{"nomad", "jobs", "statuses"}, time.Now())
+
+	namespace := args.RequestNamespace()
+	// the namespace from the UI by default is "*", but if specific jobs are
+	// requested, all with the same namespace, AllowNsOp() below may be able
+	// to quickly deny the request if the token lacks permissions for that ns,
+	// rather than iterating the whole jobs table and filtering out every job.
+	if len(args.Jobs) > 0 {
+		nses := set.New[string](0)
```

Comment:
Incredibly minor optimization: we know because we have non-zero jobs that we always have at least one namespace. Setting the initial value to 0 allocates the map and then _always_ immediately has to go allocate the extra space anyways. We can save a malloc call by just giving us a minimal size up front:

```suggestion
		nses := set.New[string](1)
```